themes/mongodb/src/js/componentEcosystemLinkTrack.js

```diff
@@ -0,0 +1,19 @@
+import {reportAnalytics, toArray} from './util';
+
+function reportClick(anchorElement) {
+    reportAnalytics('Link Clicked', {
+        'text': anchorElement.text,
+        'href': anchorElement.href,
+    })
+}
+
+export function setup() {
+    // only on ecosystem homepage, track link clicks
+    if (document.body.dataset && document.body.dataset.project === 'ecosystem' && window.location.href === 'https://docs.mongodb.com/ecosystem/') {
+        // get links on ecosystem
+        const anchors = toArray(document.getElementsByTagName('a'));
```

Comment:
We can actually do this, I think:

```
const anchors = document.querySelectorAll('a.external');
for(let i = 0; i < anchors.length; i += 1) {
   ...
}
```

This avoids creating a copy of the list of elements (as toArray does), and only iterates over external links rather than _all_ links